pkg/pgmodel/sql_bench_test.go

```diff
@@ -0,0 +1,262 @@
+package pgmodel
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+const (
+	labelCount      = 5
+	metricName      = "benchmark_metric"
+	otherMetricName = "other_benchmark_metric"
+)
+
+func BenchmarkGetSeriesIDForKeyValueArrayExistingSeries(b *testing.B) {
+	b.StopTimer()
+	withDB(b, "bench_1", func(db *pgxpool.Pool, t testing.TB) {
+		err := createMetricTableName(db, metricName)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		keys, values := generateKeysAndValues(labelCount, "label")
+		err = getSeriesIDForKeyValueArray(db, metricName, keys, values)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		var bench *testing.B
+		var ok bool
+		if bench, ok = t.(*testing.B); !ok {
+			t.Fatal("Not a benchmarking instance, stopping benchmark")
+		}
+
+		bench.ResetTimer()
+		bench.StartTimer()
+		for n := 0; n < b.N; n++ {
+			err = getSeriesIDForKeyValueArray(db, metricName, keys, values)
```

Comment:
don't we want this going through N different series?

Reply:
Again, why? Benchmark is about using existing series. Do you think it would give us better insight into the performance if we run the function on multiple series twice and measuring the second run? If so, I'll update it.